R/PopulationHR.R

```diff
@@ -0,0 +1,83 @@
+#' Calculate Population Hazard Ratios
+#'
+#' Calculates hazard ratios marginalised over subject specific random effects
+#'
+#' @param object ([`JointModelSamples`]) \cr samples as drawn from a Joint Model.
+#' @param HR_terms (`character`) \cr to include in the hazard ratio calculation.
+#' By default this include the terms used in the survival model.
+#' Set to `NULL` not include any terms
+#' @param arm (`logical`) \cr calculate hazard ratio for `arm`.
+#' @param baseline (`string`) \cr terms to model baseline hazard using variable `"time"`.
+#' Default is a B-spline from [splines]: `"bs(time, df=10)"`
+#'
+#' @returns A matrix
+#'
+#' @importFrom splines bs
+
+PopulationHR <- function(object,
+                         HR_terms = attr(terms(object@data@survival@formula), "term.labels"),
```

Comment:
Idea: take a `formula` here instead of the terms? It could default to the object's formula again.